AwarenessApisSampleKotlin/app/src/main/java/com/android/example/awarenessapissamplekotlin/MainActivity.kt

```diff
@@ -37,6 +55,83 @@ class MainActivity : AppCompatActivity() {
 
     fun onClickSnapshot(view: View) {
         Log.d(TAG, "onClickSnapshot()")
+
+        if (activityRecognitionPermissionApproved()) {
+            requestSnapshot()
+        } else {
+            requestPermissionLauncher.launch(Manifest.permission.ACTIVITY_RECOGNITION)
+        }
+    }
+
+    private fun requestSnapshot() {
+        Log.d(TAG, "requestSnapshot()")
+
+        val task: Task<DetectedActivityResponse> =
+            Awareness.getSnapshotClient(this).detectedActivity
+
+        task.addOnCompleteListener { taskResponse: Task<DetectedActivityResponse> ->
```

Comment:
can this be simplified to:
`task.addOnCompleteListener { taskResponse: ->`?

Reply:
I'd like to keep this one, as when devs are reading online, it helps them know the exact class.

Why? I find sometimes I'm just looking for the class to reference the kdocs, so I can see what the class is and any other methods I might be interested... then I don't have to clone the whole thing and try building it.